Sources/TrustlySDK/TrustlyView.swift

```
@@ -590,124 +576,108 @@ extension TrustlyView {
+                                        
+                    getTrustlySettingsWith(establish: establish) { trustlySettings in
                         
-                        DispatchQueue.main.async {
-                            // Update the UI on the main thread
-                            self.establishASWebAuthentication(onReturn: onReturn, onCancel: onCancel)
-                        }
+                        guard let settings = trustlySettings?.settings else { return }
                         
-                    } else {
-                        self.establishWebView(onReturn: onReturn, onCancel: onCancel)
+                        self.prepareEstablish(establishData: establish, inAppBrowser: settings.isInAppBrowserEnabled())
+                        
+                        loadLightbox(establish: self.establishData!, url: environment.url) { (data, response, error) in
+                            
+                            if error == nil, let url = response?.url, let data = data {
+                                DispatchQueue.main.async {
+                                    if settings.isInAppBrowserEnabled() {
+                                        self.establishASWebAuthentication(url: url, onReturn: onReturn, onCancel: onCancel)
+                                    
+                                    } else {
+                                        self.establishWebView(data: data, url: url, onReturn: onReturn, onCancel: onCancel)
+                                        
+                                    }
+                                }
+                                
+                            } else {
+                                self.cancelHandler!(self, [:])
+                                
+                            }
+                        }
                     }
+                    
+                } catch NetworkError.invalidUrl {
+                    print("Error: Invalid url.")
```

I recommend using the log class instead of print statements. Ideally, we should propagate this errors to the application.

---

NVM. The log class is only available in v4.0